test/gitStash.spec.js

```diff
@@ -0,0 +1,477 @@
+import path from 'path'
+import fsp from 'fs-promise'
+import tmp from 'tmp'
+import gitflow from '../src/gitWorkflow'
+
+jest.dontMock('execa')
+tmp.setGracefulCleanup()
+
+let wcDir
+let wcDirPath
+let gitOpts = { cwd: 'test/__fixtures__' }
+const initialContent = `module.exports = {
+    test: 'test2'
+}
+`
+
+async function gitStatus(opts = gitOpts) {
+  return gitflow.execGit(['status', '--porcelain'], opts)
+}
+
+async function gitStashList(opts = gitOpts) {
+  return gitflow.execGit(['stash', 'list'], opts)
+}
+
+async function readFile(filepath, dir = wcDirPath) {
+  const content = await fsp.readFile(path.join(dir, filepath), { encoding: 'utf-8' })
+  return content
+}
+
+describe('git', () => {
+  beforeEach(async () => {
+    wcDir = tmp.dirSync({ unsafeCleanup: true })
+    wcDirPath = wcDir.name
+    gitOpts = {
+      cwd: wcDirPath,
+      gitDir: path.join(wcDirPath, '.git')
+    }
+
+    // Init repository
+    await gitflow.execGit('init', gitOpts)
+    // Create JS file
+    await fsp.writeFile(
+      path.join(wcDirPath, 'test.js'),
+      `module.exports = {
+  test: 'test',
+
+  foo: 'bar'
+}
+`
+    )
+    await fsp.writeFile(
+      path.join(wcDirPath, 'test.css'),
+      `.test {
+    border: 1px solid green;
+}
+`
+    )
+    await gitflow.execGit(['config', 'user.name', '"test"'], gitOpts)
+    await gitflow.execGit(['config', 'user.email', '"test@test.com"'], gitOpts)
+    // Track all files
+    await gitflow.execGit(['add', '.'], gitOpts)
+    // Create inital commit
+    await gitflow.execGit(['commit', '-m', '"Initial commit"'], gitOpts)
+    // Update one of the files
+    await fsp.writeFile(path.join(wcDirPath, 'test.css'), '.test { border: red; }')
+    // Update one of the files
+    await fsp.writeFile(path.join(wcDirPath, 'test.js'), initialContent)
+  })
+
+  afterEach(() => {
+    wcDir.removeCallback()
+  })
+
+  describe('getFilesStatus', () => {
+    it('should return empty array if no changed files', async () => {
+      await gitflow.execGit(['checkout', '.'], gitOpts)
+      const res = await gitflow.getFilesStatus(gitOpts)
+      expect(res).toMatchSnapshot()
+    })
+
+    it('should return an empty array for only modified files', async () => {
+      const res = await gitflow.getFilesStatus(gitOpts)
+      expect(res).toMatchSnapshot()
+    })
+
+    it('should return an array of partially staged files', async () => {
```

Comment:
I don't think this test is written correctly. The snapshot output:
```js
exports[`git getFilesStatus should return an array of partially staged files 1`] = `
Array [
  "M  test.css",
  "MM test.js",
]
`;
```

Reply:
Removed and replaced code with https://github.com/luftywiranda13/g-status